File: codes/Question3_2/Python/NameCardMain.py
import ArrayListModule as alm
import NameCardModule as ncm

arlist = alm.ArrayList()

# 3명의 정보 저장
arlist.LInsert(ncm.NameCard("바름", "010-1234-5678"))
arlist.LInsert(ncm.NameCard("유연", "010-9123-4567"))
arlist.LInsert(ncm.NameCard("재현", "010-8912-3456"))

# 특정 한 명의 이름 탐색 및 정보 출력
data = arlist.LFirst()
if data[0] is True:
    if data[1].NameCompare("바름") is True:
        data[1].ShowNameCardInfo()
    
    data = arlist.LNext()
    while data[0] is True:
        if data[1].NameCompare("바름") is True:
            data[1].ShowNameCardInfo()
        data = arlist.LNext()
print()

# 특정 한 명의 이름 탐색 및 전화번호 변경
data = arlist.LFirst()
if data[0] is True:
    if data[1].NameCompare("유연") is True:
        data[1].ChangePhoneNum("010-7891-2345")
    
    data = arlist.LNext()
    while data[0] is True:
        if data[1].NameCompare("유연") is True:
            data[1].ChangePhoneNum("010-7891-2345")
        data = arlist.LNext()

# 특정 한 명의 이름 탐색 및 삭제
data = arlist.LFirst()
if data[0] is True:
    if data[1].NameCompare("재현") is True:
        arlist.LRemove()
    
    data = arlist.LNext()
    while data[0] is True:
        if data[1].NameCompare("재현") is True:
            arlist.LRemove()
        data = arlist.LNext()

# 남아있는 데이터 출력
data = arlist.LFirst()
if data[0] is True:
    data[1].ShowNameCardInfo()
    
    data = arlist.LNext()
    while data[0] is True:
        data[1].ShowNameCardInfo()
        data = arlist.LNext()
print()

File: codes/Question3_2/Java/Question3_2/NameCardMain.java
package Question3_2;

public class NameCardMain {
    public static void main(String[] args) {
        ArrayListGeneric<NameCard> list = new ArrayListGeneric<NameCard>();
        NameCard[] data = new NameCard[1];

        // 3명의 정보 저장
        list.LInsert(new NameCard("바름", "010-1234-5678"));
        list.LInsert(new NameCard("유연", "010-9123-4567"));
        list.LInsert(new NameCard("재현", "010-8912-3456"));

        // 특정 한 명의 이름 탐색 및 정보 출력
        if (list.LFirst(data)) {
            if (data[0].NameCompare("바름")) {
                data[0].ShowNameCardInfo();

                while(list.LNext(data)) {
                    if(data[0].NameCompare("바름"))
                        data[0].ShowNameCardInfo();
                }
            }
        }
        System.out.println();

        // 특정 한 명의 이름 탐색 및 전화번호 변경
        if (list.LFirst(data)) {
            if (data[0].NameCompare("유연"))
                data[0].ChangePhoneNum("010-7891-2345");

            while(list.LNext(data)) {
                if(data[0].NameCompare("유연"))
                    data[0].ChangePhoneNum("010-7891-2345");
            }
        }

        // 특정 한 명의 이름 탐색 및 삭제
        if (list.LFirst(data)) {
            if (data[0].NameCompare("재현"))
                list.LRemove();

            while(list.LNext(data))
                if(data[0].NameCompare("재현"))
                    list.LRemove();
        }

        // 남아있는 데이터 출력
        if (list.LFirst(data)) {
            data[0].ShowNameCardInfo();

            while(list.LNext(data)) {
                data[0].ShowNameCardInfo();
            }
        }
        System.out.println();
    }
}
